web/src/components/profile/ProfileErrorAlert.tsx

```diff
@@ -0,0 +1,61 @@
+import { Alert } from "@/components/njwds-extended/Alert";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { getProfileConfig } from "@/lib/domain-logic/getProfileConfig";
+import { ProfileContentField } from "@/lib/types/types";
+import { templateEval } from "@/lib/utils/helpers";
+import { ReactElement } from "react";
+
+interface Props {
+  fieldErrors: string[];
+}
+
+export const ProfileErrorAlert = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+
+  const displayProfileErrorAlert = (): boolean => props.fieldErrors.length > 0;
+
+  const getHeader = (field: string): string => {
+    return getProfileConfig({
+      config: Config,
+      fieldName: field as ProfileContentField,
+    }).header;
+  };
+
+  const errorFieldsIds = (): string[] => {
+    const errorFieldsIds: string[] = [];
+    for (const error of props.fieldErrors) {
+      if (error === "willSellPetCareItems" && !errorFieldsIds.includes("petCareHousing")) {
+        errorFieldsIds.push("petCareHousing");
+      } else if (getHeader(error)) {
+        errorFieldsIds.push(error);
+      }
+    }
+    return errorFieldsIds;
+  };
+
+  const profileErrorAlertText = (): string =>
+    errorFieldsIds().length === 1
+      ? templateEval(Config.profileDefaults.default.profileErrorAlert, {
+          fieldText: Config.profileDefaults.default.profileErrorAlertOneField,
+        })
+      : templateEval(Config.profileDefaults.default.profileErrorAlert, {
+          fieldText: Config.profileDefaults.default.profileErrorAlterMultipleFields,
+        });
+
+  return (
+    <>
+      {displayProfileErrorAlert() && (
+        <>
+          <Alert variant="error" dataTestid={"profile-error-alert"}>
+            <div>{profileErrorAlertText()}</div>
+            <ul>
+              {errorFieldsIds().map((id) => (
+                <li key={id}>{getHeader(id)}</li>
+              ))}
+            </ul>
+          </Alert>
+        </>
+      )}
+    </>
+  );
```

Comment:
[dust] There's nothing wrong with this, but I probably would have written this return differently. The main reason being that It's more explicit that nothing gets displayed when `displayProfileErrorAlert()` returns `false`. This is just preference though.

There is an additional React fragment in here you can remove, but that's just me nit picking.
```suggestion
return displayProfileErrorAlert() ? (
  <Alert variant="error" dataTestid={"profile-error-alert"}>
    <div>{profileErrorAlertText()}</div>
    <ul>
      {errorFieldsIds().map((id) => (
        <li key={id}>{getHeader(id)}</li>
      ))}
    </ul>
  </Alert>
) : <></>
```

Reply:
Separate thought, we had previously talked about how we should separate the UI logic from the conditional logic on whether it should display. I think we agreed it should be defined at the parent level. Therefore, `fieldErrors` does not need to be a prop.

Reply:
I disagree, I'd rather see this as a guard clause

Reply:
```
if (!displayProfileErrorAlert()) return null;
return <Alert>...
```